File: 01/__main__.py
with open('./input', 'r') as f:
    elves = [[]]
    for line in f:
        if line == '\n':
            elves.append([])
        else:
            elves[-1].append(int(line.strip()))

srtd = sorted([sum(elf) for elf in elves])
print('part 1:', srtd[-1])
print('part 2:', sum(srtd[-1:-4:-1]))

File: 01/rust/src/main.rs
pub fn main() {
    let elves = include_str!("../../input").split("\n\n");
    let mut calorie_totals = vec![];
    for elf in elves {
        let total_calories = elf
            .split('\n')
            .take_while(|l| !l.is_empty())
            .fold(0, |a, b| a + b.parse::<i32>().unwrap());
        calorie_totals.push(total_calories);
    }
    calorie_totals.sort_by(|a, b| b.cmp(a));
    println!("part 1: {}", calorie_totals[0]);
    println!("part 2: {}", calorie_totals[..3].iter().sum::<i32>());
}
